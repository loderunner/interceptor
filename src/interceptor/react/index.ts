export { Provider as InterceptProvider, useIntercepts } from './intercept'
