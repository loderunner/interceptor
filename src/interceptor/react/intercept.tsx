import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import * as intercept from '../intercept'

interface InterceptContextType {
  intercepts: intercept.Intercept[]
  addIntercept: (pattern: string) => void
  removeIntercept: (inter: intercept.Intercept) => void
}

const InterceptContext = createContext<InterceptContextType>({
  intercepts: [],
  addIntercept: () => {},
  removeIntercept: () => {},
})

interface Props {
  children: React.ReactNode
}

export const Provider = ({ children }: Props) => {
  const [intercepts, setIntercepts] = useState([...intercept.intercepts])

  const addIntercept = (pattern: string) => {
    intercept.addIntercept({ pattern, enabled: true })
    setIntercepts([...intercept.intercepts])
  }
  const removeIntercept = (inter: intercept.Intercept) => {
    intercept.removeIntercept(inter)
    setIntercepts([...intercept.intercepts])
  }

  return (
    <InterceptContext.Provider
      value={{ intercepts, addIntercept, removeIntercept }}
    >
      {children}
    </InterceptContext.Provider>
  )
}

export const useIntercepts = () => useContext(InterceptContext)
