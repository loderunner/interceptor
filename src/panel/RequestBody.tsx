import * as ContentType from 'content-type'
import * as React from 'react'
import { useMemo } from 'react'

import { JSONBodyView } from './components/body-view/JSON'

import type { Request } from '@/interceptor'

interface Props {
  request: Request
}

interface Props {
  request: Request
}

const RequestBody = ({ request }: Props) => {
  const contentType = useMemo(() => {
    try {
      const contentTypeHeader = Object.entries(request.headers).find(
        ([headerName]) => headerName.toLowerCase() === 'content-type'
      )
      if (contentTypeHeader === undefined) {
        return ''
      }

      const contentType = ContentType.parse(contentTypeHeader[1])
      return contentType.type
    } catch (err) {
      return ''
    }
  }, [request])

  if (request.postData) {
    try {
      if (contentType === 'application/json' || contentType === 'text/plain') {
        JSON.parse(request.postData)
      }
      return <JSONBodyView jsonData={request.postData} />
    } catch (err) {
      // ignore error parsing JSON
    }
  }

  return (
    <pre className="bg-slate-100 px-6 py-6 whitespace-pre-wrap break-all">
      {request.postData}
    </pre>
  )
}

export default RequestBody
