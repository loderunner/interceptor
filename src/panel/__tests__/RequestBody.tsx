import { cleanup, render } from '@testing-library/react'
import * as React from 'react'
import '@testing-library/jest-dom'

import RequestBody from '../RequestBody'

jest.mock('@/interceptor/hooks')

describe('[RequestBody]', () => {
  afterEach(() => {
    cleanup()
  })

  it('should match initial snapshot', () => {
    const { container } = render(
      <RequestBody requestId={globalMocks.request.id} />
    )
    expect(container).toMatchSnapshot()
  })
})
