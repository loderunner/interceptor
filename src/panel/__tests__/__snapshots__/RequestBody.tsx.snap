// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RequestBody] should match initial snapshot 1`] = `
<div>
  <div
    class="mt-8 flex flex-col justify-items-stretch "
  >
    <div
      class="space-y-1"
    >
      <div
        class="flex items-baseline space-x-1"
        style="margin-left: 0rem;"
      >
        <pre
          class="text-amber-700"
        >
          hello
          :
        </pre>
         
        <div
          class="flex-auto"
        >
          <pre
            class="whitespace-pre-wrap break-all text-sky-400"
          >
            "world"
          </pre>
        </div>
      </div>
      <div
        class="flex items-baseline space-x-1"
        style="margin-left: 0rem;"
      >
        <pre
          class="text-amber-700"
        >
          obj
          :
        </pre>
         
        <pre
          class="flex items-center space-x-0.5"
        >
          {
          <button
            class="mx-0.5 rounded-sm bg-slate-400 hover:bg-slate-500 active:bg-slate-600"
          >
            <svg />
          </button>
          }
        </pre>
      </div>
      <div
        class="space-y-1"
      >
        <div
          class="flex items-baseline space-x-1"
          style="margin-left: 1rem;"
        >
          <pre
            class="text-amber-700"
          >
            barbes
            :
          </pre>
           
          <div
            class="flex-auto"
          >
            <pre
              class="whitespace-pre-wrap break-all text-sky-400"
            >
              "rochechouart"
            </pre>
          </div>
        </div>
        <div
          class="flex items-baseline space-x-1"
          style="margin-left: 1rem;"
        >
          <pre
            class="text-amber-700"
          >
            version
            :
          </pre>
           
          <div
            class="flex-auto"
          >
            <pre
              class="whitespace-pre-wrap break-all text-rose-500"
            >
              1
            </pre>
          </div>
        </div>
        <div
          class="flex items-baseline space-x-1"
          style="margin-left: 1rem;"
        >
          <pre
            class="text-amber-700"
          >
            toto
            :
          </pre>
           
          <div
            class="flex-auto"
          >
            <pre
              class="whitespace-pre-wrap break-all text-purple-500"
            >
              null
            </pre>
          </div>
        </div>
        <div
          class="flex items-baseline space-x-1"
          style="margin-left: 1rem;"
        >
          <pre
            class="text-amber-700"
          >
            arr
            :
          </pre>
           
          <pre
            class="flex items-center space-x-0.5"
          >
            [
            <button
              class="mx-0.5 rounded-sm bg-slate-400 hover:bg-slate-500 active:bg-slate-600"
            >
              <svg />
            </button>
            ]
          </pre>
        </div>
        <div
          class="space-y-1 hidden"
        >
          <div
            class="flex items-baseline space-x-1"
            style="margin-left: 2rem;"
          >
            <pre
              class="text-amber-700"
            >
              0
              :
            </pre>
             
            <div
              class="flex-auto"
            >
              <pre
                class="whitespace-pre-wrap break-all text-rose-500"
              >
                0.7309955249202127
              </pre>
            </div>
          </div>
          <div
            class="flex items-baseline space-x-1"
            style="margin-left: 2rem;"
          >
            <pre
              class="text-amber-700"
            >
              1
              :
            </pre>
             
            <div
              class="flex-auto"
            >
              <pre
                class="whitespace-pre-wrap break-all text-rose-500"
              >
                0.45152102178420384
              </pre>
            </div>
          </div>
          <div
            class="flex items-baseline space-x-1"
            style="margin-left: 2rem;"
          >
            <pre
              class="text-amber-700"
            >
              2
              :
            </pre>
             
            <div
              class="flex-auto"
            >
              <pre
                class="whitespace-pre-wrap break-all text-rose-500"
              >
                0.3101491368469109
              </pre>
            </div>
          </div>
          <div
            class="flex items-baseline space-x-1"
            style="margin-left: 2rem;"
          >
            <pre
              class="text-amber-700"
            >
              3
              :
            </pre>
             
            <div
              class="flex-auto"
            >
              <pre
                class="whitespace-pre-wrap break-all text-rose-500"
              >
                0.33964078594296154
              </pre>
            </div>
          </div>
          <div
            class="flex items-baseline space-x-1"
            style="margin-left: 2rem;"
          >
            <pre
              class="text-amber-700"
            >
              4
              :
            </pre>
             
            <div
              class="flex-auto"
            >
              <pre
                class="whitespace-pre-wrap break-all text-rose-500"
              >
                0.579201406715314
              </pre>
            </div>
          </div>
        </div>
        <div
          class="flex items-baseline space-x-1"
          style="margin-left: 1rem;"
        >
          <pre
            class="text-amber-700"
          >
            nested
            :
          </pre>
           
          <pre
            class="flex items-center space-x-0.5"
          >
            {
            <button
              class="mx-0.5 rounded-sm bg-slate-400 hover:bg-slate-500 active:bg-slate-600"
            >
              <svg />
            </button>
            }
          </pre>
        </div>
        <div
          class="space-y-1 hidden"
        >
          <div
            class="flex items-baseline space-x-1"
            style="margin-left: 2rem;"
          >
            <pre
              class="text-amber-700"
            >
              thx
              :
            </pre>
             
            <div
              class="flex-auto"
            >
              <pre
                class="whitespace-pre-wrap break-all text-rose-500"
              >
                1138
              </pre>
            </div>
          </div>
          <div
            class="flex items-baseline space-x-1"
            style="margin-left: 2rem;"
          >
            <pre
              class="text-amber-700"
            >
              star
              :
            </pre>
             
            <div
              class="flex-auto"
            >
              <pre
                class="whitespace-pre-wrap break-all text-sky-400"
              >
                "wars"
              </pre>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex items-baseline space-x-1"
        style="margin-left: 0rem;"
      >
        <pre
          class="text-amber-700"
        >
          done
          :
        </pre>
         
        <div
          class="flex-auto"
        >
          <pre
            class="whitespace-pre-wrap break-all text-green-500"
          >
            true
          </pre>
        </div>
      </div>
    </div>
  </div>
</div>
`;
